starlette_cramjam/middleware.py

```diff
@@ -126,13 +127,14 @@ async def send_with_compression(self, message: Message) -> None:
             body = message.get("body", b"")
             more_body = message.get("more_body", False)
 
-            self.buffer.write(self.compressor.compress(body))
+            self.compressor.compress(body)
             if not more_body:
+                self.buffer.write(self.compressor.finish())
                 message["body"] = self.buffer.getvalue()
                 self.buffer.close()
                 await self.send(message)
                 return
-
+            self.buffer.write(self.compressor.flush())
             message["body"] = self.buffer.getvalue()
             self.buffer.seek(0)
             self.buffer.truncate()
```

Comment:
Note, I didn't change the `self.buffer` but it seems like you don't need it? After each `flush` it's truncated and after `.finish` it's closed, so maybe more efficient just to use the output directly. (and  get the bytes view by wrapping the output with `bytes(...)`

Reply:
🙏 I'll do it in another PR